### ISSUE TITLE
[BUG] Type error using ignorePatterns

### ISSUE THREAD

**Issue by jrdx0** [halvardssm/oak-middleware-jwt#5]
**Describe the bug**
Trying use code example of ignorePatterns returns an error

**Screenshots or error log**
```
error: TS2322 [ERROR]: Type '(string | RegExp | { path: string; methods: string[]; })[]' is not assignable to type 'IgnorePath[]'.
  Type 'string | RegExp | { path: string; methods: string[]; }' is not assignable to type 'IgnorePath'.
    Type '{ path: string; methods: string[]; }' is not assignable to type 'IgnorePath'.
      Type '{ path: string; methods: string[]; }' is not assignable to type 'Pattern'.
        Types of property 'methods' are incompatible.
          Type 'string[]' is not assignable to type 'HTTPMethods[]'.
            Type 'string' is not assignable to type 'HTTPMethods'.

The expected type comes from property 'ignorePatterns' which is declared here on type 'JwtMiddlewareOptions' ignorePatterns?: Array<IgnorePath>;
at https://raw.githubusercontent.com/halvardssm/oak-middleware-jwt/master/src/jwtMiddleware.ts:35:3
```

**System (please complete the following information):**
- OS: [Windows 10]
- Deno version: [1.4.6]
- Module version [0.5.0]

**Additional context**
This is the complete code:
``` typescript
import { jwtMiddleware } from "https://raw.githubusercontent.com/halvardssm/oak-middleware-jwt/master/mod.ts"
import { Middleware, Application } from "https://deno.land/x/oak@v6.0.2/mod.ts";
 
const app = new Application();
 
const ignorePatterns = ["/baz", /buz/, {path: "/biz", methods: ["GET"]}]
 
app.use(jwtMiddleware<Middleware>({key: "foo", algorithm: "HS512", ignorePatterns}));

await app.listen({port: 8000})    
```

**Comment by halvardssm** [comment]
Hi @stillalivx ! I have tried to recreate your issue and I found the problem, and its laying with typescript. If you add the type to the const `ignorePatterns`, it works fine. bellow you can find your modified code 😊 

Let me know if you experience any other issues!

```ts
import { IgnorePath, jwtMiddleware } from "https://raw.githubusercontent.com/halvardssm/oak-middleware-jwt/master/mod.ts"
import { Middleware, Application } from "https://deno.land/x/oak@v6.0.2/mod.ts";
 
const app = new Application();
 
const ignorePatterns: IgnorePath[] = ["/baz", /buz/, {path: "/biz", methods: ["GET"]}]
 
app.use(jwtMiddleware<Middleware>({key: "foo", algorithm: "HS512", ignorePatterns}));

await app.listen({port: 8000})    
```

**Comment by halvardssm** [comment]
I just published a new version, `1.0.0`. `IgnorePath` is now `IgnorePattern`. Hope you find this module useful!

**Comment by jrdx0** [comment]
Thanks! Now everything work fine!
